metricbeat/helper/prometheus/textparse.go

```diff
@@ -0,0 +1,694 @@
+				lookupMetricName = metricName[:len(metricName)-6]
+			} else {
+				lookupMetricName = metricName
+			}
+		case textparse.MetricTypeGauge:
+			var gauge = &Gauge{Value: &v}
+			metric = &OpenMetric{Name: &metricName, Gauge: gauge, Label: labelPairs}
+			lookupMetricName = metricName
+		case textparse.MetricTypeInfo:
+			value := int64(v)
+			var info = &Info{Value: &value}
+			metric = &OpenMetric{Name: &metricName, Info: info, Label: labelPairs}
+			lookupMetricName = metricName
+		case textparse.MetricTypeSummary:
+			lookupMetricName, metric = summaryMetricName(metricName, v, lset.Get(model.QuantileLabel), lbls.String(), &t, summariesByName)
+			metric.Label = labelPairs
+			if !isSum(metricName) {
+				continue
+			}
+			metricName = lookupMetricName
+		case textparse.MetricTypeHistogram:
+			if hasExemplar := parser.Exemplar(&e); hasExemplar {
+				exm = &e
+			}
+			lookupMetricName, metric = histogramMetricName(metricName, v, lset.Get(labels.BucketLabel), lbls.String(), &t, false, exm, histogramsByName)
+			metric.Label = labelPairs
+			if !isSum(metricName) {
+				continue
+			}
+			metricName = lookupMetricName
+		case textparse.MetricTypeGaugeHistogram:
+			if hasExemplar := parser.Exemplar(&e); hasExemplar {
+				exm = &e
+			}
+			lookupMetricName, metric = histogramMetricName(metricName, v, lset.Get(labels.BucketLabel), lbls.String(), &t, true, exm, histogramsByName)
+			metric.Label = labelPairs
+			metric.Histogram.IsGaugeHistogram = true
+			if !isGSum(metricName) {
+				continue
+			}
+			metricName = lookupMetricName
+		case textparse.MetricTypeStateset:
+			value := int64(v)
+			var stateset = &Stateset{Value: &value}
+			metric = &OpenMetric{Name: &metricName, Stateset: stateset, Label: labelPairs}
+			lookupMetricName = metricName
+		case textparse.MetricTypeUnknown:
+			var unknown = &Unknown{Value: &v}
+			metric = &OpenMetric{Name: &metricName, Unknown: unknown, Label: labelPairs}
+			lookupMetricName = metricName
+		default:
+			lookupMetricName = metricName
+		}
+
+		fam, ok = metricFamiliesByName[lookupMetricName]
+		if !ok {
+			fam = &MetricFamily{Type: mt}
+			metricFamiliesByName[lookupMetricName] = fam
+		}
+
+		fam.Name = &metricName
+
+		if hasExemplar := parser.Exemplar(&e); hasExemplar && mt != textparse.MetricTypeHistogram && metric != nil {
+			if !e.HasTs {
+				e.Ts = t
+			}
+			metric.Exemplar = &e
+		}
+
+		if tp != nil && metric != nil {
+			t = *tp
+			metric.TimestampMs = &t
+		}
+
+		fam.Metric = append(fam.Metric, metric)
+	}
+
+	families := make([]*MetricFamily, 0, len(metricFamiliesByName))
+	for _, v := range metricFamiliesByName {
+		if v.Metric != nil {
+			families = append(families, v)
+		}
+	}
+	return families, nil
+}
+
+const (
+	TextVersion     = "0.0.4"
+	OpenMetricsType = `application/openmetrics-text`
+
+	// The Content-Type values for the different wire protocols.
+	FmtUnknown string = `<unknown>`
```

Comment:
Nit. This value looks a bit arbitrary, is this something supported by textparse? Could we use the zero value here?
```suggestion
	FmtUnknown string = ``
```

Reply:
It's coming from https://github.com/prometheus/common/blob/main/expfmt/expfmt.go#L30 and was used inside https://github.com/prometheus/common/blob/main/expfmt/decode.go#L64 that we were using so better to keep it as is so as to not break any compatability.

Reply:
Ok